src/components/DateFilter.tsx

```diff
@@ -0,0 +1,218 @@
+      const monthFrom = dateFrom.toLocaleString('default', { month: 'short' });
+      dateFromString = `${dayRange.from.day} ${monthFrom} ${dayRange.from.year}`;
+    }
+
+    if (dayRange.to) {
+      const dateTo = new Date(dayRange.to.year, dayRange.to.month - 1, dayRange.to.day);
+      const monthTo = dateTo.toLocaleString('default', { month: 'short' });
+      dateToString = `${dayRange.to.day} ${monthTo} ${dayRange.to.year}`;
+    }
+
+    return dateFromString && dateToString
+      ? `${dateFromString} - ${dateToString}`
+      : dateFromString
+      ? `${dateFromString} -`
+      : '';
+  };
+
+  const renderCustomInput = ({ ref }: RenderInputProps) => {
+    return (
+      <input
+        ref={ref as RefObject<HTMLInputElement>} // necessary
+        placeholder={
+          dateRange && dateRange === ALL_TIMES ? dateRangeToString(dateRange) : `Select Date Range`
+        }
+        value={dayRange && dayRange.from && dayRange.to ? `${formatDateRange(dayRange)}` : ''}
+        className='custom-date-filter-input'
+      />
+    );
+  };
+
+  const resetDateFilter = () => {
+    setDayRange({ from: null, to: null });
+    exploreUrl?.setDateRange('AllTimes');
+  };
+
+  const handleChange = (dayRange: DayRange) => {
+    const formattedDateRange = formatDateRangeForUrl(dayRange.from, dayRange.to);
+    if (formattedDateRange) exploreUrl?.setDateRange(formattedDateRange);
+    setDayRange(dayRange);
+  };
+
+  const formatDateRangeForUrl = (dateFrom: DayValue, dateTo: DayValue): DateRange | undefined => {
+    if (dateFrom && dateTo) {
+      // e.g:  dateFrom=2021-06-22&dateTo=2021-09-22
+      const from = new Date(dateFrom.year, dateFrom.month - 1, dateFrom.day).toISOString();
+      const to = new Date(dateTo.year, dateTo.month - 1, dateTo.day).toISOString();
+      return `from=${dayjs(from).format('YYYY-MM-DD')}&to=${dayjs(to).format('YYYY-MM-DD')}` as DateRange;
+    }
+  };
+
+  const myCustomLocale: Locale = {
+    // months list by order
+    months: [
+      'January',
+      'February',
+      'March',
+      'April',
+      'May',
+      'June',
+      'July',
+      'August',
+      'September',
+      'October',
+      'November',
+      'December',
+    ],
+
+    // week days by order
+    weekDays: [
+      {
+        name: 'Monday',
+        short: 'M',
+      },
+      {
+        name: 'Tuesday',
+        short: 'T',
+      },
+      {
+        name: 'Wednesday',
+        short: 'W',
+      },
+      {
+        name: 'Thursday',
+        short: 'T',
+      },
+      {
+        name: 'Friday',
+        short: 'F',
+      },
+      {
+        name: 'Saturday',
+        short: 'S',
+        isWeekend: true,
+      },
+      {
+        name: 'Sunday', // used for accessibility
+        short: 'S', // displayed at the top of days' rows
+        isWeekend: true, // is it a formal weekend or not?
+      },
+    ],
+
+    // just play around with this number between 0 and 6
+    weekStartingIndex: 0,
```

Comment:
Can we set this to Monday, please? :)

Reply:
Actually this file is for the first version of the date filter, the one that we discarded. But I will check how to do that for the new data range selector (the one with from and to input boxes).